docs/_running_external.rst

```diff
@@ -0,0 +1,223 @@
+output YAML file. In this case, both inputs are datasets rather than collections
+or parameters, so they are both classified as ``class: File``. A placeholder
+path is also included, which you should change to the paths of your chosen
+input files.
+
+Another more complex example, also including a collection and parameter as
+input, might look like the following:
+
+::
+
+    input_dataset:
+      class: File
+      path: todo_test_data_path.ext
+    input_collection:
+      class: Collection
+      collection_type: list
+      elements:
+      - class: File
+        identifier: todo_element_name
+        path: todo_test_data_path.ext
+    input_parameter: param_todo
+
+
+For the collection, each dataset is listed, with a path and identifier specified.
+For the parameter, only a single value has to be specified, but ensure it is of
+the correct type, e.g. float, if that is what the workflow requires. 
+
+Using workflow and dataset IDs
+===============================================
+
+If you ran all the commands above then you probably noticed that both the
+workflow and the input datasets get newly uploaded at each execution. If you
+want to run the same workflow multiple times, you may prefer to avoid this.
+In the examples given so far, all workflows and datasets are specified by means
+of a local path, but Planemo also allows you to use the IDs created by Galaxy
+as well. These IDs are unique to each Galaxy server, so this approach isn't
+transferrable if you want to run your workflows on multiple servers.
+
+The first step is to ensure all the datasets which are required for the
+workflow are already uploaded. You can either do this by running the workflow
+once in the normal way, as described above, or just manually uploading through
+the web interface.
+
+To get dataset IDs, you can click on the dataset's 'View details' button (a
+small letter 'i' in a circle). This provides various information about the
+dataset and the job which created it. Under the 'Job information' section,
+there is a row named 'History Content API ID'. For each input dataset, copy
+this string (it will probably look something like ``457d46215431cc37baf96108ad87f351``)
+and paste it into the workflow job file so it looks something like the following:
+
+::
+
+    GSE37268_mof3:
+      class: File
+      galaxy_id: "457d46215431cc37baf96108ad87f351"
+    Genes:
+      class: File
+      galaxy_id: "55f30adf41ae36455431abeaa185ed89"
+
+
+i.e. just replace the ``path`` line with ``galaxy_id``.
+
+You can do exactly the same with a collection; either of the following will
+work:
+
+::
+
+    input_collection2:
+      class: Collection
+      galaxy_id: "9d362c51f575db89"
+    input_collection2:
+      class: Collection
+      collection_type: list
+      elements:
+      - class: File
+        identifier: todo_element_name
+        galaxy_id: "457d46215431cc37baf96108ad87f351"
+
+
+In the first example, an existing collection will be used (by specifying its
```

Comment:
first example might be confusing, maybe saying explicitly int the input_collection2 example  ... and maybe this should be collection1?